lib/src/type_utils.dart

```diff
@@ -12,9 +12,9 @@ extension DartTypeExtension on DartType {
     if (self is InterfaceType) {
       return self.element2;
     } else if (self is NeverType) {
-      return self.element as TypeDefiningElement;
+      return self.element2 as TypeDefiningElement;
     } else if (self is TypeParameterType) {
-      return self.element;
+      return self.element2;
     } else {
       // Remaining cases like `DynamicType`, `FunctionType`, and `VoidType`.
```

Comment:
BTW, we had an issue with `dynamic`, with this kind of rollback, you still can use `DynamicType.element2`.

Reply:
Ah ok, it looks like it is not needed. But I'll keep it in mind.